docs/source/pynq_libraries/pynq_microblaze_subsystem.rst

```diff
@@ -64,4 +64,36 @@ More information about the PynqMicroblaze class, and its API can be found in the
 PynqMicroBlaze class, and further example notebooks can be found in those
 sections.
 
-   
+Creating a New PYNQ Microblaze
+------------------------------
+
+Any hierarchy that contains a Microblaze and meets the above requirements of
+having AXI-accessible code memory, a PS interrupt line and a reset line can be
+used in PYNQ. However in order to use your Microblaze with the IPython magic
+you must provide a board support package or BSP. BSPs are generated from Xilinx
+SDK and contain all of the drivers and configuration data for the peripherals
+attached to the Microblaze.
+
+PYNQ provides a TCL script to generate the BSP from the hardware description
+file which can be found in the ``boards/sw_repo`` directory of the repository,
+along with the drivers needed for Python/C communication and the ``pynqmb``
+hardware abstraction library. Creating and using the BSP requires the following
+steps:
+
+1. Export Hardware from Vivado to generate the HDF file
+
+2. In the ``boards/sw_repo`` directory run ``xsct build_xsdk.tcl $HDF_FILE hw_project``
+
```

Comment:
I think we can run `make` here since we have a makefile (looks better than the `xsct` call?)

Reply:
`make` will only recreate the BSPs for the base overlay though. I wanted to explain how people could create BSPs for their own custom instances. I accept that we are missing the part about running `make` to create the base overlay BSPs though.

Reply:
Or we can call `make HDF=$HDF` which I agree looks better